File: Naheulbook.Web/Responses/GroupResponse.cs
using System.Collections.Generic;
using Naheulbook.Shared.TransientModels;
using Newtonsoft.Json.Linq;

// ReSharper disable AutoPropertyCanBeMadeGetOnly.Global
// ReSharper disable UnusedMember.Global
// ReSharper disable UnusedAutoPropertyAccessor.Global

namespace Naheulbook.Web.Responses
{
    public class GroupResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public JObject? Data { get; set; }
        public GroupConfig Config { get; set; } = null!;

        public IList<int> CharacterIds { get; set; } = null!;
        public IList<GroupGroupInviteResponse> Invites { get; set; } = null!;
    }
}

File: src/app/api/responses/group-response.ts
import {GroupGroupInviteResponse} from './group-group-invite-response';
import {IGroupConfig} from '../shared';

export interface GroupResponse {
    id: number;
    name: string;
    data?: any;
    config: IGroupConfig;
    characterIds: number[];
    invites: GroupGroupInviteResponse[];
}
